src/include/distributed/subplan_execution.h

```diff
@@ -19,5 +19,20 @@ extern int SubPlanLevel;
 
 extern void ExecuteSubPlans(DistributedPlan *distributedPlan);
 
+/**
+ * IntermediateResultsHashEntry is used to store which nodes need to receive
+ * intermediate results. Given an intermediate result name, you can lookup
+ * the list of nodes that can possibly run a query that will use the
+ * intermediate results.
+ *
+ * The nodeIdList contains a set of unique WorkerNode ids that have placements
+ * that can be used in non-colocated subquery joins with the intermediate result
+ * given in the key.
+ */
+typedef struct IntermediateResultsHashEntry
+{
+	char *key;
```

Comment:
actually, it's probably more convenient to put `char key[NAMEDATALEN];` here

Otherwise you would have to not use HASH_ELEM and allocate IntermediateResultsHashEntry yourself.

Reply:
yes, that seems more common, I was writing a similar comment :) 

Check `plperl_query_entry` @hanefi for an example